cmd/limits.go

```diff
@@ -0,0 +1,117 @@
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/exoscale/egoscale"
+	"github.com/spf13/cobra"
+)
+
+type LimitsItemOutput struct {
+	Resource string `json:"resource"`
+	Used     int    `json:"used"`
+	Max      int    `json:"max"`
+}
+
+type LimitsOutput []LimitsItemOutput
+
+func (o *LimitsOutput) toJSON()  { outputJSON(o) }
+func (o *LimitsOutput) toText()  { outputText(o) }
+func (o *LimitsOutput) toTable() { outputTable(o) }
+
+var limitsCmd = &cobra.Command{
+	Use:   "limits",
+	Short: "Current account limits",
+	Long: fmt.Sprintf(`This command lists the safety limits currently enforced on your account.
+
+Supported output template annotations: %s`,
+		strings.Join(outputterTemplateAnnotations(&LimitsOutput{}), ", ")),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return output(listLimits())
+	},
+}
+
+func listLimits() (outputter, error) {
+	limits, err := cs.ListWithContext(gContext, &egoscale.ResourceLimit{})
+	if err != nil {
+		return nil, err
+	}
+
+	display := map[string]string{
+		"user_vm":           "Instances",
+		"snapshot":          "Snapshots",
+		"template":          "Templates",
+		"public_elastic_ip": "IP Addresses",
+		"network":           "Private Networks",
+	}
+
+	out := LimitsOutput{}
+
+	for _, key := range limits {
+		limit := key.(*egoscale.ResourceLimit)
+
+		if display[limit.ResourceTypeName] != "" {
+			rUsed, err := fetchUsedResources(limit.ResourceTypeName)
+			if err != nil {
+				return nil, err
+			}
+
+			out = append(out, LimitsItemOutput{
+				Resource: display[limit.ResourceTypeName],
+				Max:      int(limit.Max),
+				Used:     rUsed,
+			})
+		}
+	}
+
+	return &out, nil
+}
+
+func fetchUsedResources(resourceType string) (int, error) {
```

Comment:
(untested) 

```suggestion
func fetchUsedResources(resourceType string) (int, error) {
	switch resourceType {
	case "user_vm":
		instances, err := cs.ListWithContext(gContext, &egoscale.VirtualMachine{})
		if err != nil {
			return 0, err
		}

		return len(instances), nil

	case "snapshot":
		snapshots, err := cs.ListWithContext(gContext, &egoscale.Snapshot{})
		if err != nil {
			return 0, err
		}

		return len(snapshots), nil

	case "template":
		templates, err := cs.ListWithContext(gContext, &egoscale.Template{})
		if err != nil {
			return 0, err
		}

		return len(templates), nil

	case "network":
		networks, err := cs.ListWithContext(gContext, &egoscale.Network{})
		if err != nil {
			return 0, err
		}

		return len(networks), nil

	default: // public_elastic_ip
		eips, err := cs.ListWithContext(gContext, &egoscale.IPAddress{IsElastic: true})
		if err != nil {
			return 0, err
		}

		return len(eips), nil
	}
}

```

Reply:
No, because as you can see, there is more "resourceType" than we need:
[ResourceTypeName](https://github.com/exoscale/egoscale/blob/21b8548dd15850ce490130dc8f2aad57dcdc1a44/resource_limits.go#L8)

so in your case, if a resource is not in the display variable will be set to the `default` case

Reply:
ok.